lib/domain/entities/system_permission.dart

```diff
@@ -1 +1,39 @@
-enum SystemPermissionStatus { always, denied, whileInUse, permanentlyDenied }
+import 'package:freezed_annotation/freezed_annotation.dart';
+import 'package:movna/domain/faults.dart';
+
+part 'system_permission.freezed.dart';
+
+enum SystemPermissionStatus {
+  always,
+  denied,
+  whileInUse,
+  permanentlyDenied;
+
+  bool get isGranted => switch (this) {
+        SystemPermissionStatus.always => true,
+        SystemPermissionStatus.whileInUse => true,
+        _ => false,
+      };
+}
+
+/// Data class containing a [SystemPermissionStatus] as well as a boolean
+/// indicating if permission was requested or not.
+///
+/// [demanded] represents the business logic need of the permission.
+///
+/// If [demanded] is `true` then [status] should not be null.
+/// if [failure] is not null and [demanded] is true then it means an error
+/// occurred while getting the permission.
+@freezed
```

Comment:
```suggestion
@freezed
class SystemPermissionStatusHolder with _$SystemPermissionStatusHolder {
  const factory SystemPermissionStatusHolder({
    /// Null when not demanded yet.
    required Result<SystemPermissionStatus, Fault>? status,
  }) = _SystemPermissionHolder;
  
  SystemPermissionHolder._();
  
  /// Whether the permission was already demanded or not.
  bool get demanded => this.status != null;
  static const SystemPermissionStatusHolder notDemanded =
      SystemPermissionStatusHolder(
    status: null,
  );
}
```

Reply:
Also a question :
Why do you sometimes use factories that return a specific instance, like `GpsCoordinates.paris`, but sometimes a static instance, is there a reason to prefer one or the other ? As a C++ developer I would always put static instances, but in dart I may also write named constructors without any parameter (removing the `factory` keyword in the `GpsCoordinates.paris` example)

Reply:
the factory VS static field is a lack of consistency on my part, I think we should use static fields as they are `const`

And I'll see what I can do about the fields of the holder

Reply:
I removed the permission holder altogether, it didn't have much sense to keep it just to store a single `Result`